### PR TITLE
no http timeout

### PR DESCRIPTION
**Pull request by ngalayko**
With this option, response from server was 5sec minimum, now it's asap

If I misunderstood and did something bad, let me know

### CODE REVIEW

**Comment by yfronto** [comment]
Thanks for the PR. Actually, http.Client.Timeout does the opposite. It doesn't set the minimum; it sets the maximum time a client will wait. 

From initial request, through redirects and downloads and everything, if 5 seconds have passed, then Request.Cancel is called to kill the connection.

I wouldn't want to supply a library that doesn't have a timeout. If that behavior is really desired, the client could be overridden. You can read more from the [net/http docs][https://golang.org/pkg/net/http/#Client].

**Comment by ngalayko** [comment]
It is odd, because with this option, i coundn't get response faster than 5sec

Anyway, thx for package, this little problem seems to appear in order of mine project feature, where every request is initial

**Comment by yfronto** [comment]
I should probably make the client in the Client object public so it can be overridden. If I don't get to it, I'd be happy to accept a PR that does that, then you could set the timeout in your code.

**Comment by yfronto** [comment]
The build fails because you'll need to update http_hepler.go to use the new name. You can follow the links to TravisCI to see the log of the failure, but it should be failing with `go build` locally as well.

File: main.go [hunk 1]
```diff
@@ -11,15 +11,19 @@ package newrelic
 import (
 	"net/http"
 	"net/url"
+	"time"
 )
 
 // DefaultAPIURL is the default base URL for New Relic's latest API.
-const DefaultAPIURL = "https://api.newrelic.com/v2/"
+const (
+	DefaultAPIURL = "https://api.newrelic.com/v2/"
```
Comment (yfronto) [inline main.go]: These const values should remain private.

File: main.go [hunk 2]
```diff
@@ -31,7 +35,7 @@ func NewClient(apiKey string) *Client {
 	}
 	return &Client{
 		apiKey:     apiKey,
-		httpClient: &http.Client{},
+		HttpClient: &http.Client{Timeout: DefaultTimeout},
```
Comment (yfronto) [inline main.go]: You should create another func:

```go
func NewClient(apiKey string, httpClient *http.Client) *Client {
...
}
```

That way, a consumer can choose to override by providing their own client. This function can call that one.

**Comment by yfronto** [comment]
Does 0fb206c fix your issue? It allows you to supply your own http.Client, so you can override any default in the client.

**Comment by ngalayko** [comment]
i have fixed it but in local repo, i don't have ability to push here (last time was by github web client)
i will do it, but a bit later

**Comment by ngalayko** [comment]
oh, i see you changes, thx a lot, i think some of people might need it

**Comment by yfronto** [comment]
Thanks for the suggestion, there are definitely plenty of cases for override.

**Pull request closed without merging**